drivers/wifi/nrfwifi/off_raw_tx/src/off_raw_tx_api.c

```diff
@@ -0,0 +1,430 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @brief File containing API definitions for the Offloaded raw TX feature.
+ */
+
+#include <ctype.h>
+#include <stdlib.h>
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/drivers/wifi/nrfwifi/off_raw_tx/off_raw_tx_api.h>
+#include <off_raw_tx.h>
+#include <fmac_api.h>
+
+#define DT_DRV_COMPAT nordic_wlan
+LOG_MODULE_DECLARE(wifi_nrf, CONFIG_WIFI_NRF70_LOG_LEVEL);
+
+struct nrf_wifi_off_raw_tx_drv_priv off_raw_tx_drv_priv;
+extern const struct nrf_wifi_osal_ops nrf_wifi_os_zep_ops;
+
+static const int valid_rate[] = { 1, 2, 55, 11, 6, 9, 12, 18, 24, 36, 48, 54,
+				  0, 1, 2, 3, 4, 5, 6, 7, -1 };
+
+/* DTS uses 1dBm as the unit for TX power, while the RPU uses 0.25dBm */
+#define MAX_TX_PWR(label) DT_PROP(DT_NODELABEL(nrf70), label) * 4
+static void configure_tx_pwr_settings(struct nrf_wifi_tx_pwr_ctrl_params *ctrl_params,
```

Comment:
can't this be reused instead of copying?

Reply:
It's part of fmac_main file which we are excluding in compilation.